src/js/common/schemaform/helpers.js

```diff
@@ -445,25 +449,25 @@ export function removeHiddenData(schema, data) {
       if (typeof data[next] !== 'undefined') {
         const nextData = removeHiddenData(schema.properties[next], data[next]);
 
+        // if the data was removed, then just unset it
         if (typeof nextData === 'undefined') {
           return _.unset(next, current);
         }
+
+        // if data was updated (like a nested prop was removed), update it
+        if (nextData !== data[next]) {
```

Comment:
This code is in the `updatedSchema.type === 'object'` case. The diff is making it look like it isn't, I think.